aws-android-sdk-auth-userpools/src/main/java/com/amazonaws/mobile/auth/userpools/SignUpActivity.java

```diff
@@ -94,13 +100,16 @@ public void signUp(final View view) {
         Log.d(LOG_TAG, "email = " + email);
         Log.d(LOG_TAG, "phone = " + phone);
 
+
+        final Integer minimumPasswordLength = CognitoUserPoolsSignInProvider.getMinimumPasswordLength(configuration);
+
         if (username.isEmpty()) {
             showError(getString(R.string.sign_up_username_missing));
             return;
         }
 
-        if (password.length() < 6) {
-            showError(getString(R.string.password_length_validation_failed));
+        if (minimumPasswordLength != null && password.length() < minimumPasswordLength) {
```

Comment:
when would minimum be null? Should we fall back to 6 as a safety? Is lower than 6 ever allowed?

Reply:
If the password requirements are not specified in `awsconfiguration` this can be null.

6 is the minimum in Cognito, but if we don't know what the customer has set as their minimum then we just don't want to validate client-side, since any value we choose is likely to be wrong. The original customer report complains that validating with the wrong length leads to a confusing end user experience.

So if we don't know what the minimum is we'll just let the server side validate the password and display whatever error message they return if it's too short (this part is existing functionality).